### PR TITLE
Update User Guide based on in class review

### PR DESCRIPTION
**Pull request by RyanCheungJF**
https://ryancheungjf.github.io/tp/UserGuide.html to view (I took down my site temporarily as there was an issue with workflows jn)

https://github.com/AY2122S2-CS2103T-T09-2/tp/issues/112

Edited one Diagram just now in Developer Guide in order to trigger the workflow, but turns out Github workflow was down. It's a future change I wanted to make for DG so I'll just keep it in this PR for now.

Solves:
- Target Audience in front matters
- Legend on icons and notations
- Improve tone to less formal and more personal
- Improve ambiguity of explanations(i.e. “e.g. 5 will not delete if there are only 4 applicants in the list” -> For this example, it is unclear what 5 represents).
- Add flag command to table of cmds and usage
- Change single line code blocks to Multiple line without side scroll(PDF conversion scroll sideways

Could possibly have better explanation for flag and un flag in the future
Search is not updated yet

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -77,156 +105,254 @@ Format: `help`
...
 ```
-search n/NAME
+search [n/NAME]... [j/JOB]... [r/ROUND]... [s/SKILL]...
 ```
 * The search is case-insensitive. e.g `hans` will match `Hans`
-* Only the name is searched.
-* Only full words will be matched e.g. `Han` will not match `Hans`
 * First name/Last name will be matched e.g. `Han` will match `Han Lee`
+* Only full words will be matched e.g. `Han` will not match `Hans`
+* Search will not work with multiple prefixes. e.g `search n/Alex j/Interview` will only yield results of `search n/Alex`
```
Comment (kaiyichen) [inline docs/UserGuide.md:231]: which version of search are we suppose to include?

File: docs/UserGuide.md [hunk 2]
```diff
@@ -77,156 +105,254 @@ Format: `help`
...
+
+<br>
 
 Format:
 ```
 add n/NAME p/PHONE_NUMBER e/EMAIL j/JOB r/ROUND [s/SKILLS]...
 ```
 <div markdown="span" class="alert alert-primary">:bulb: **Tip:**
-A person can have any number of skills (including 0)
+A person can have any number of skills (including none at all)
```
Comment (kaiyichen) [inline docs/UserGuide.md:123]: Is it better to state it as an applicant?
```suggestion
An applicant can have any number of skills (including none at all)
```

File: docs/UserGuide.md [hunk 3]
```diff
@@ -77,156 +105,254 @@ Format: `help`
...
 ```
 * The search is case-insensitive. e.g `hans` will match `Hans`
-* Only the name is searched.
-* Only full words will be matched e.g. `Han` will not match `Hans`
 * First name/Last name will be matched e.g. `Han` will match `Han Lee`
+* Only full words will be matched e.g. `Han` will not match `Hans`
+* Search will not work with multiple prefixes. e.g `search n/Alex j/Interview` will only yield results of `search n/Alex`
+
+<div markdown="span" class="alert alert-primary">:bulb: **Tip:**
+You can try searching for multiple names in the applicant list. Try `search n/Alex Bernice` to look up these 2 applicants!
```
Comment (kaiyichen) [inline docs/UserGuide.md:234]: is this better?
```suggestion
You can try searching for multiple names in the applicant list. Try `search n/Alex Bernice` to search for applicants with name 'Alex' or 'Bernice'!
```

File: docs/UserGuide.md [hunk 4]
```diff
@@ -77,156 +105,254 @@ Format: `help`
...
+Skills: [Accounting][Equities][Cryptocurrency]
 ```
 
 [Back to top <img src="images/back-to-top-icon.png" width="25px" />](#table-of-contents)
 
 ---
 ### Deleting an applicant : `delete`
 
-Deletes a specific job applicant, specified by index.
+Allows you to delete a specific job applicant, specified by index.
```
Comment (kaiyichen) [inline docs/UserGuide.md:330]: seems like 'applicant' is used in other features, probably better to standardize it?
```suggestion
Allows you to delete a specific applicant, specified by index.
```

**Review by kaiyichen — COMMENTED** [review]
LGTM!! left some suggestions

File: docs/UserGuide.md (hunk 1 above)
Comment (RyanCheungJF) [reply docs/UserGuide.md:231]: Yup I have updated UG to work with your latest PR.

File: docs/UserGuide.md (hunk 2 above)
Comment (RyanCheungJF) [reply docs/UserGuide.md:123]: Good spot, thanks!

File: docs/UserGuide.md (hunk 4 above)
Comment (RyanCheungJF) [reply docs/UserGuide.md:330]: Fixed!

File: docs/UserGuide.md (hunk 3 above)
Comment (RyanCheungJF) [reply docs/UserGuide.md:234]: Thanks for the suggestion, will add it in.

**Review by ryantanlien — APPROVED** [review]
LGTM!

**Pull request merged**